concordium-consensus/src/Concordium/ImportExport.hs

```diff
@@ -127,160 +191,123 @@ writeSection ::
     BlockHash ->
     -- |Height of first block in section
     BlockHeight ->
-    -- |Handle to write to
+    -- |Export directory
+    FilePath ->
+    -- |Index file handle
     Handle ->
-    -- |Action to write out the blocks and return the number of them
-    m Word64 ->
-    -- |Action to write out the finalization records and return the number of them
-    m Word64 ->
+    -- |Chunk size
+    Word64 ->
     m ()
-writeSection
+writeChunks
     sectionGenesisIndex
     sectionProtocolVersion
     sectionGenesisHash
     sectionFirstBlockHeight
-    hdl
-    writeBlocks
-    writeFinRecs = do
+    outDir indexHdl chunkSize = do
+        let chunkName = outDir </> "blocks-"
+                        ++ show sectionGenesisIndex ++ "-"
+                        ++ show sectionFirstBlockHeight
+                        ++ ".dat"
+        chunkHdl <- liftIO $ initialHandle chunkName
+
         (sectionStart, blocksStart) <- liftIO $ do
-            sectionStart <- hTell hdl
+            BS.hPut chunkHdl $ encode (3 :: Version)
+            sectionStart <- hTell chunkHdl
             -- Write a dummy section header that we will later overwrite
-            runPutH (liftPut $ putWord64be sectionHeaderLength >> put placeholderSectionHeader) hdl
-            blocksStart <- hTell hdl
+            runPutH (liftPut $ putWord64be sectionHeaderLength >> put placeholderSectionHeader) chunkHdl
+            blocksStart <- hTell chunkHdl
             return (sectionStart, blocksStart)
-        sectionBlockCount <- writeBlocks
-        blocksEnd <- liftIO $ hTell hdl
-        sectionFinalizationCount <- writeFinRecs
+        -- how many blocks would it take to fill the exported chunk to its `chunkSize`
+        -- (non-zero means we're out of blocks to export in the current section)
+        (chunkUnused, finRecsMissing) <- exportBlocksToChunk chunkHdl sectionFirstBlockHeight chunkSize
```

Comment:
It seems it would be cleaner to return the number of blocks that were exported instead of this chunkUnused.